p2p/src/authenticated/discovery/mod.rs

```diff
@@ -95,6 +95,13 @@
 //! channels. Each registered channel ([Sender], [Receiver]) handles its own message queuing
 //! and rate limiting.
 //!
+//! ## Compression
+//!
+//! Stream compression is purposely avoided in this implementation to prevent known
```

Comment:
I'd be more explicit about this being an attack on **encrypted** streams that use compression.

For someone first looking, I wouldn't come to that conclusion.